refinery/core/models.py

```diff
@@ -1711,53 +1726,62 @@ def attach_outputs_dataset(self):
                     workflow_output=output_connection.name
                 )
             )
-            # retrieve uuid of corresponding output file if exists
-            logger.info("Results for '%s' and %s.%s: %s",
-                        self.uuid, output_connection,
-                        output_connection.filetype, analysis_result)
-
-            derived_data_file_node.file_uuid = analysis_result.file_store_uuid
-
-            logger.debug("Output file %s ('%s') assigned to node %s ('%s')",
-                         output_connection, analysis_result.file_store_uuid,
-                         derived_data_file_node.name,
-                         derived_data_file_node.uuid)
-
+            if output_connection.is_refinery_file:
+                # retrieve uuid of corresponding output file if exists
+                logger.info(
+                    "Results for '%s' and %s.%s: %s",
+                    self.uuid,
+                    output_connection,
+                    output_connection.filetype,
+                    analysis_result
+                )
+                derived_data_file_node.file_uuid = (
+                    analysis_result.file_store_uuid
+                )
+                logger.debug(
+                    "Output file %s ('%s') assigned to node %s ('%s')",
+                    output_connection,
+                    analysis_result.file_store_uuid,
+                    derived_data_file_node.name,
+                    derived_data_file_node.uuid
+                )
             output_connection.node = derived_data_file_node
             output_connection.save()
+
             # get graph edge that corresponds to this output node:
             # a. attach output node to source data transformation node
             # b. attach output node to target data transformation node
             # (if exists)
             if len(graph.edges([output_connection.step])) > 0:
                 for edge in graph.edges_iter([output_connection.step]):
-                    output_id = "{}_{}".format(
-                        output_connection.step,
-                        output_connection.filename
-                    )
+                    output_id = output_connection.get_output_connection_id()
```

Comment:
hmm: I was imagining just one method, `get_id`, instead of separate `get_output_connection_id` and `get_input_connection_id`?

Reply:
Good call, thats less confusing